lib/league_statistics.rb

```diff
@@ -0,0 +1,100 @@
+require_relative "./stat_tracker"
+require "csv"
+
+class LeagueStatistics < StatTracker
+
+  def count_of_teams
+    total_games = CSV.read(@teams, :headers=>true)
+    total_games.count
+  end
+
+  def team_name(id)
+    rows = CSV.read(@teams, :headers => true, :header_converters => :symbol)
+    rows.find do |row|
+      return row[:teamname] if row[:team_id] == id.to_s
+    end
+  end
+
+  def team_scores
+    scores("home_and_away")
+  end
+
+  def visitor_scores
+    scores("away")
+  end
+
+  def home_team_scores
+    scores("home")
+  end
+
+  def scores(side)
```

Comment:
I noticed that there was overlap in the methods for helper methods team_scores, visitor_scores and home_team_scores. Moved things around and created this helper method to refactor the code and make it cleaner. scores is a method that takes a side which is either "home", "away", or "home_and_away" to create this hash with team_id's as the key and the scores of all their games